Panic in RedisQueue on Terminate

When set 
```ini
[queue.task]
TYPE = redis
CONN_STR = "addrs=127.0.0.1:6379 db=0"

[queue.mail]
TYPE = redis
CONN_STR = "addrs=127.0.0.1:6379 db=0"
```

and then stop gitea

```log
panic: close of nil channel

goroutine 63 [running]:
code.gitea.io/gitea/modules/queue.(*RedisQueue).Terminate(0xc00036b730)
	/Volumes/data/Projects/gitea/gitea/modules/queue/queue_redis.go:219 +0x153
code.gitea.io/gitea/modules/graceful.(*Manager).RunAtTerminate.func1(0xc00018a960, 0xc00044e400, 0x6d7e940, 0xc0000de008)
	/Volumes/data/Projects/gitea/gitea/modules/graceful/manager.go:126 +0x110
created by code.gitea.io/gitea/modules/graceful.(*Manager).RunAtTerminate
	/Volumes/data/Projects/gitea/gitea/modules/graceful/manager.go:123 +0x7a
panic: close of nil channel

goroutine 202 [running]:
code.gitea.io/gitea/modules/queue.(*RedisQueue).Terminate(0xc000148700)
	/Volumes/data/Projects/gitea/gitea/modules/queue/queue_redis.go:219 +0x153
code.gitea.io/gitea/modules/graceful.(*Manager).RunAtTerminate.func1(0xc00018a960, 0xc00198e700, 0x6d7e940, 0xc0000de008)
	/Volumes/data/Projects/gitea/gitea/modules/graceful/manager.go:126 +0x110
created by code.gitea.io/gitea/modules/graceful.(*Manager).RunAtTerminate
	/Volumes/data/Projects/gitea/gitea/modules/graceful/manager.go:123 +0x7a
```

--- a/modules/queue/queue_redis.go
+++ b/modules/queue/queue_redis.go
@@ -84,11 +84,12 @@ func NewRedisQueue(handle HandlerFunc, cfg, exemplar interface{}) (Queue, error)
 			boostWorkers:       config.BoostWorkers,
 			maxNumberOfWorkers: config.MaxWorkers,
 		},
-		queueName: config.QueueName,
-		exemplar:  exemplar,
-		closed:    make(chan struct{}),
-		workers:   config.Workers,
-		name:      config.Name,
+		queueName:  config.QueueName,
+		exemplar:   exemplar,
+		closed:     make(chan struct{}),
+		terminated: make(chan struct{}),
+		workers:    config.Workers,
+		name:       config.Name,
 	}
 	if len(dbs) == 0 {
 		return nil, errors.New("no redis host specified")

